models/articles-model.js

```diff
@@ -8,18 +8,27 @@ exports.selectArticleById = (article_id) => {
   });
 };
 
-exports.selectArticles = () => {
-  const queryString = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes,
-  CAST(COUNT(comments.article_id) AS INT) AS comment_count
-  FROM articles
-  LEFT JOIN comments ON comments.article_id = articles.article_id
-  GROUP BY comments.article_id, articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes
-  ORDER BY created_at DESC
-  `;
-
-  return db.query(queryString).then(({ rows }) => {
-    return rows;
-  });
+exports.selectArticles = (topic) => {
+  let queryString = "";
+
+  if (topic) {
+    const queryString = `SELECT * FROM articles WHERE topic = $1  ORDER BY created_at DESC`;
+
+    return db.query(queryString, [topic]).then(({ rows }) => {
+      return rows;
+    });
+  } else {
+    const queryString = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes,
+    CAST(COUNT(comments.article_id) AS INT) AS comment_count
+    FROM articles
+    LEFT JOIN comments ON comments.article_id = articles.article_id 
+    GROUP BY comments.article_id, articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes 
+    ORDER BY created_at DESC`;
+
+    return db.query(queryString).then(({ rows }) => {
+      return rows;
+    });
```

Comment:
As it looks like the model is already checking if the topic is fine, you can condense some of your controller logic as line 14 is checking if the parameter is undefined or not. You'll still need your check exists in here though.